### PR TITLE
Add Virtual Network to ProjectCore resource group

### PR DESCRIPTION
**Pull request by robbiepaul**
At the moment we're not using the virtual network for anything, however we intend to use it for either protecting resources with a firewall or splitting out worker processes (both require our own virtual network)

Virtual networks are [free of charge](https://azure.microsoft.com/en-us/pricing/details/virtual-network/).

We add the virtual network first as we expect to rely on these in the later resource group deployments.

### CODE REVIEW

**Comment by erbridge** [comment]
[CYBER 7: Protect critical infrastructure with a firewall](https://trello.com/c/tJNwUwtt/560-cyber-7-protect-critical-infrastructure-with-a-firewall)

File: azure/resource_groups/ProjectCore/template.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        },
+        "subnets": [
+          {
+            "name": "default",
+            "properties": {
+              "addressPrefix": "10.211.0.0/27",
+              "serviceEndpoints": [],
+              "delegations": [],
+              "privateEndpointNetworkPolicies": "Enabled",
+              "privateLinkServiceNetworkPolicies": "Enabled"
```
Comment (chelnak) [inline azure/resource_groups/ProjectCore/template.json]: I'd recommend using service endpoints and delegations here. Private link has only just come out in preview. I'd probably avoid this until it's GA.

The block template above allows you to pass in a subnet configuration object. You can include service endpoint and delegation config within that object.

For ref: https://docs.microsoft.com/en-us/azure/templates/microsoft.network/2019-04-01/virtualnetworks/subnets#subnetpropertiesformat-object

File: azure/resource_groups/ProjectCore/template.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]"
+    }
+  },
+  "variables": {
+    "vNetName": "[concat(parameters('resourceNamePrefix'), '-vn')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Network/virtualNetworks",
```
Comment (chelnak) [inline azure/resource_groups/ProjectCore/template.json]: Can we use the building block here?

https://github.com/DFE-Digital/bat-platform-building-blocks/blob/master/templates/virtual-network.json

It's already being used for ACI integration elsewhere.

Comment (erbridge) [reply azure/resource_groups/ProjectCore/template.json]: Oh, nice call. I assumed there wouldn't be one, yet. Thanks.

File: azure/resource_groups/ProjectCore/template.json (hunk 1 above)
Comment (erbridge) [reply azure/resource_groups/ProjectCore/template.json]: Yeah, we decided to leave those out to be added when we need them for specific uses.

**Pull request merged**